src/components/TodoContext.jsx

```diff
@@ -0,0 +1,132 @@
+import { createContext, useState, useEffect } from "react";
+
+export const TodoContext = createContext();
+
+export function TodoProvider({ children }) {
+  const [todos, setTodos] = useState([]);
+
+  const onAdd = async function addTodo(newTodo) {
+    try {
+      const response = await fetch(
+        "https://jsonplaceholder.typicode.com/todos",
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(newTodo),
+        }
+      );
+      const data = await response.json();
+
+      const completeTodo = {
+        ...newTodo,
+        id: self.crypto.randomUUID(),
+        userId: data.id,
+      };
+
+      const updatedTodos = [...todos, completeTodo];
+      setTodos(updatedTodos);
+
+      // Save the updated todos to local storage
+      saveToLocalStorage(updatedTodos);
```

Comment:
Where is this function defined?

Reply:
Please also cleanup: remove `locaStorage` from all other components where it is used (`TODO` and `IntputToDo`)